modules/core/src/test/scala/ch/epfl/bluebrain/nexus/admin/core/projects/ProjectsSpec.scala

```diff
@@ -59,145 +57,165 @@ class ProjectsSpec
-  private val aggProject                              = MemoryAggregate("projects")(Initial, next, EvalProject().apply).toF[Future]
+  private val orgsAggregate                           = MemoryAggregate("organizations")(Initial, next, Eval().apply).toF[Future]
+  private val aggProject                              = MemoryAggregate("projects")(Initial, next, Eval().apply).toF[Future]
   private val cl                                      = mock[SparqlClient[Future]]
   implicit val shaclValidator: ShaclValidator[Future] = ShaclValidator(ImportResolver.noop[Future])
-  private val projects                                = Projects(aggProject, cl)
+  private val organizations                           = Organizations(orgsAggregate, cl)
+  private val projects                                = Projects(organizations, aggProject, cl)
 
   override implicit val patienceConfig: PatienceConfig =
     PatienceConfig(5 seconds, 100 milliseconds)
-
   trait Context {
-    val id: ProjectReference = genReference()
-    val value: ProjectValue  = genProjectValue()
+    val id: ProjectReference = genProjectReference()
+    val value: Json          = genProjectValue()
+
+    val orgValue: Json = genOrganizationValue()
   }
 
   "A Project bundle" should {
-    implicit val hasWrite: HasWriteProjects =
-      applyRef[HasWriteProjects](
-        FullAccessControlList((Identity.Anonymous(), Path./, Permissions(Read, Permission("projects/write"))))
-      ).toPermTry.success.value
-    implicit val hasOwn: HasCreateProjects =
-      applyRef[HasCreateProjects](FullAccessControlList(
-        (Identity.Anonymous(), Path./, Permissions(Read, Permission("projects/create"))))).toPermTry.success.value
     implicit val hasRead: HasReadProjects =
       applyRef[HasReadProjects](FullAccessControlList(
         (Identity.Anonymous(), Path./, Permissions(Read, Permission("projects/read"))))).toPermTry.success.value
 
     "create a new project" in new Context {
-      projects.create(id, value.asJson).futureValue shouldEqual RefVersioned(id, 1L)
-      projects.fetch(id).futureValue shouldEqual Some(Project(id, 1L, value, value.asJson, deprecated = false))
+      organizations.create(id.organizationReference, orgValue).futureValue
+      projects.create(id, value).futureValue shouldEqual RefVersioned(id, 1L)
+
+      val proj = projects.fetch(id).futureValue.get
+      proj.id.value shouldEqual id
+      proj.uuid should fullyMatch regex """[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}"""
+      proj.deprecated shouldEqual false
+      proj.rev shouldEqual 1L
+      proj.value shouldEqual value
+
     }
 
     "prevent creating a project without a name" in new Context {
+      organizations.create(id.organizationReference, orgValue).futureValue
       val rej =
         projects.create(id, value.asJson.removeKeys("name")).failed.futureValue.asInstanceOf[CommandRejected].rejection
       rej shouldBe a[ResourceRejection.ShapeConstraintViolations]
     }
 
```

Comment:
missing test for `prevent updating a project with maxAttachmentSize as a string`